tests/frontier/opcodes/test_calldataload.py

```diff
@@ -0,0 +1,116 @@
+"""test `CALLDATALOAD` opcode."""
+
+import pytest
+
+from ethereum_test_forks import Byzantium, Fork
+from ethereum_test_tools import Account, Alloc, Bytecode, StateTestFiller, Transaction
+from ethereum_test_tools.vm.opcode import Opcodes as Op
+
+
+@pytest.mark.parametrize(
+    "mstore,args_size,code_for_address_a,tx_data,address_a_storage",
+    [
+        (
+            (Op.MSTORE8(offset=0x0, value=0x25) + Op.MSTORE8(offset=0x1, value=0x60)),
+            0x2,
+            (Op.PUSH1[0x0] + Op.CALLDATALOAD + Op.PUSH1[0x0] + Op.SSTORE),
+            b"\x00",
+            Account(
+                storage={0x00: 0x2560000000000000000000000000000000000000000000000000000000000000}
+            ),
+        ),
+        (
+            (
+                Op.MSTORE(
+                    offset=0x0,
+                    value=0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF,
+                )
+                + Op.MSTORE8(offset=0x20, value=0x23)
+            ),
+            0x21,
+            (Op.PUSH1[0x1] + Op.CALLDATALOAD + Op.PUSH1[0x0] + Op.SSTORE + Op.STOP),
+            b"\x01",
+            Account(
+                storage={0x00: 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF23}
+            ),
+        ),
+        (
+            (
+                Op.MSTORE(
+                    offset=0x0,
+                    value=0x123456789ABCDEF0000000000000000000000000000000000000000000000000,
+                )
+                + Op.MSTORE8(offset=0x20, value=0x0)
+                + Op.MSTORE8(offset=0x21, value=0x24)
+            ),
+            0x22,
+            (Op.PUSH1[0x5] + Op.CALLDATALOAD + Op.PUSH1[0x0] + Op.SSTORE + Op.STOP),
+            b"\x02",
+            Account(
+                storage={0x00: 0xBCDEF00000000000000000000000000000000000000000000000000024000000}
+            ),
+        ),
+    ],
+    ids=[
+        "two_bytes",
+        "word_n_byte",
+        "34_bytes",
+    ],
+)
+def test_calldataload(
+    state_test: StateTestFiller,
+    mstore: Bytecode,
+    args_size: int,
+    code_for_address_a: Bytecode,
+    fork: Fork,
+    tx_data: bytes,
+    pre: Alloc,
+    address_a_storage: Account,
+):
+    """
+    Test `CALLDATALOAD` opcode.
+
+    Based on https://github.com/ethereum/tests/blob/ae4791077e8fcf716136e70fe8392f1a1f1495fb/src/GeneralStateTestsFiller/VMTests/vmTests/calldatacopyFiller.yml
+    """
+    address_a = pre.deploy_contract(code_for_address_a)
```

Comment:
It might be unnecessary to get the entire code as a parameter since it's very similar in all cases, so we could change it only having a `calldataload_offset` parameter and change this to:
```suggestion
    address_a = pre.deploy_contract(Op.SSTORE(0, Op.CALLDATALOAD(calldataload_offset)) + Op.STOP)
```